### PR TITLE
Backend foreign key

### PR DESCRIPTION
**Pull request by norkans7**
Includes #102

### CODE REVIEW

**Comment by nicpottier** [comment]
Looks good but why did you end up switching to a foreign key here? Just curious, did you run into issues with your other design?

**Comment by norkans7** [comment]
I did not yet have an issue, just heard that this is the wrong way in case the value needs to change
and figured out I could use the backend object as an attribute on the syncers and do not query the db many times

File: dash_test_runner/testapp/migrations/0003_auto_20180405_1238.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('testapp', '0002_auto_20180312_1302'),
+    ]
+
+    operations = [
+        migrations.AlterField(
```
Comment (norkans7) [inline dash_test_runner/testapp/migrations/0003_auto_20180405_1238.py]: Altering the field, since I think nobody uses this model

**Comment by nicpottier** [comment]
k.. ya, the other bit on the query is that you can just do a prefetch_related instead, which won't do a join and is plenty quick.

**Comment by norkans7** [comment]
I have select_related on the syncer https://github.com/rapidpro/ureport/pull/350/files#diff-147c0a6b9dc99ef5e7b430d62be6fdbeR32

is prefetch faster that select_related ?

**Comment by nicpottier** [comment]
It is a bit simpler for the database since there is no join. (prefetch is done at the django level instead) If there are very few unique things that are related such as this then doing a prefetch probably makes sense.

**Pull request merged**